features/devices/isis/isis_route_install_test.go

```diff
@@ -0,0 +1,317 @@
+//go:build all || feature || b2b
+
+package isis
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/open-traffic-generator/snappi/gosnappi"
+	"github.com/open-traffic-generator/tests/helpers/otg"
+)
+
+func TestIsIsRouteInstall(t *testing.T) {
+
+	testConst := map[string]interface{}{
+		"pktRate":      int64(50),
+		"pktCount":     int32(100),
+		"pktSize":      int32(128),
+		"txMac":        "00:00:01:01:01:01",
+		"txIp":         "1.1.1.1",
+		"txGateway":    "1.1.1.2",
+		"txPrefix":     int32(24),
+		"txIpv6":       "1100::1",
+		"txv6Gateway":  "1100::2",
+		"txv6Prefix":   int32(64),
+		"rxMac":        "00:00:01:01:01:02",
+		"rxIp":         "1.1.1.2",
+		"rxGateway":    "1.1.1.1",
+		"rxPrefix":     int32(24),
+		"rxIpv6":       "1100::2",
+		"rxv6Gateway":  "1100::1",
+		"rxv6Prefix":   int32(64),
+		"txRouteCount": int32(1),
+		"rxRouteCount": int32(1),
+		"txAdvRouteV4": "10.10.10.1",
+		"rxAdvRouteV4": "20.20.20.1",
+		"txAdvRouteV6": "::10:10:10:01",
+		"rxAdvRouteV6": "::20:20:20:01",
+	}
+
+	api := otg.NewOtgApi(t)
+	c := isisRouteInstallConfig(api, testConst)
+
+	fmt.Println(c.ToJson())
+	api.SetConfig(c)
+
+	api.StartProtocols()
+
+	api.WaitFor(
+		func() bool { return isisRouteInstallMetricsOk(api, testConst) },
+		&otg.WaitForOpts{FnName: "WaitForIsIsMetrics",
+			Timeout: time.Duration(20) * time.Second},
+	)
+
+	api.StartTransmit()
+
+	api.WaitFor(
+		func() bool { return isisRouteInstallFlowMetricsOk(api, testConst) },
+		&otg.WaitForOpts{FnName: "WaitForFlowMetrics"},
+	)
+}
+
+func isisRouteInstallConfig(api *otg.OtgApi, tc map[string]interface{}) gosnappi.Config {
+	c := api.Api().NewConfig()
+
+	ptx := c.Ports().Add().SetName("ptx").SetLocation(api.TestConfig().OtgPorts[0])
+	prx := c.Ports().Add().SetName("prx").SetLocation(api.TestConfig().OtgPorts[1])
+
+	c.Layer1().Add().
+		SetName("ly").
+		SetPortNames([]string{ptx.Name(), prx.Name()}).
+		SetSpeed(gosnappi.Layer1SpeedEnum(api.TestConfig().OtgSpeed))
+
+	tx := c.Devices().Add().SetName("tx")
+	rx := c.Devices().Add().SetName("rx")
+
+	// transmit
+	txEth := tx.Ethernets().
+		Add().
+		SetName("txEth").
+		SetPortName(ptx.Name()).
+		SetMac(tc["txMac"].(string)).
+		SetMtu(1500)
+
+	txEth.
+		Ipv4Addresses().
+		Add().
+		SetName("txIp").
+		SetAddress(tc["txIp"].(string)).
+		SetGateway(tc["txGateway"].(string)).
+		SetPrefix(tc["txPrefix"].(int32))
+
+	txEth.
+		Ipv6Addresses().
+		Add().
+		SetName("txIpv6").
+		SetAddress(tc["txIpv6"].(string)).
+		SetGateway(tc["txv6Gateway"].(string)).
+		SetPrefix(tc["txv6Prefix"].(int32))
+
+	txIsis := tx.Isis().
+		SetSystemId("640000000001").
+		SetName("tx_isis")
```

Comment:
follow the camelCase convention for naming objects

Reply:
updated